source/module_esolver/esolver_ks_lcao.cpp

```diff
@@ -687,7 +717,9 @@ void ESolver_KS_LCAO<TK, TR>::hamilt2density(int istep, int iter, double ethr)
     // save input rho
     this->pelec->charge->save_rho_before_sum_band();
     // save density matrix for mixing
-    if (GlobalV::MIXING_RESTART > 0 && GlobalV::MIXING_DMR && iter >= this->p_chgmix->mixing_restart)
+    if (GlobalV::MIXING_RESTART > 0 
+        && GlobalV::MIXING_DMR 
+        && this->p_chgmix->mixing_restart_count > 0)
```

Comment:
I'm confused why mixing_restart_count is "int" and only need to judge it equal or larger than 0, what is your design for this parameter?

Reply:
I add some explanation.